sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -324,6 +330,8 @@ class ResolveSessionCatalog(
 
   object SessionCatalog {
     def unapply(nameParts: Seq[String]): Option[(CatalogPlugin, Seq[String])] = nameParts match {
+      case AsTemporaryViewIdentifier(i) if catalogManager.v1SessionCatalog.isTemporaryTable(i) =>
```

Comment:
Do we really need this change? AFAIK the v1 commands keep the table name and lookup table metadata at runtime. If they support views, they would lookup views at that time. It looks to me that we don't need to handle temp views when converting to v1 command.

Reply:
Yes, this is needed. Without this check, commands with temporary views will be converted to v2 plans because the v2 session catalog doesn't return a `V1Table`. Queries like `ALTER TABLE` with a known view should not be converted to v2 plans, or else we need more cases in `CheckAnalysis`.

Reply:
I ended up needing to revert this because some commands work even if the identifier is for a temp table.